### PR TITLE
Convert compile demo into a unit test.

### PR DESCRIPTION
**Pull request by NeilFraser**
Squash and merge.

### CODE REVIEW

File: tests/jsunit/db_test.js [hunk 1]
```diff
@@ -1,76 +0,0 @@
-/**
```
Comment (rachel-fenichel) [inline tests/jsunit/db_test.js]: What happened to this file?

Comment (NeilFraser) [reply tests/jsunit/db_test.js]: It was not referenced in `index.html`, meaning none of the tests were being run.  It contained a single test (`test_DB_getNeighbours`) which failed when run.  Meanwhile `connection_db_test.js` contains a test of the same name which passed when run.  `db_test.js` appears to be dead code.

**Comment by rachel-fenichel** [comment]
Travis is broken.

**Comment by NeilFraser** [comment]
The 'travis-compile' branch to merge is reliably green:
  https://travis-ci.org/google/blockly/branches
But the merge test failed to connect to the Selenium server:
  https://travis-ci.org/google/blockly/jobs/287212868

Looks unrelated to Blockly, and that Travis itself is flaky.  I'll make a dummy commit and see if this is intermittent.

**Pull request merged**